Source/Charts/Data/Implementations/Standard/CombinedChartData.swift

```diff
@@ -209,23 +209,17 @@ open class CombinedChartData: BarLineScatterCandleBubbleChartData
         return allData.index(of: data)
     }
     
-    open override func removeDataSet(_ dataSet: ChartDataSetProtocol!) -> Bool
+    open override func removeDataSet(_ dataSet: ChartDataSetProtocol) -> Element?
     {
-        let datas = allData
-        
-        var success = false
-        
-        for data in datas
+        for data in allData
         {
-            success = data.removeDataSet(dataSet)
-            
-            if success
+            if let e = data.removeDataSet(dataSet)
```

Comment:
I'm don't know. I don't like how this method was written in the first place, but I don't feel like changing it because I have a much bigger change coming for the data models.